front/src/app/todo-list/todo-list.component.ts

```diff
@@ -2,46 +2,68 @@ import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnInit, OnDestro
 import { ApiService } from '../shared/api-service/api.service';
 import { FormControl, FormGroup } from '@angular/forms';
 import { ActivatedRoute } from '@angular/router';
-import { TaskGroups } from '../shared/models/shared.models';
-import { Subscription } from 'rxjs';
+import { TaskGroupsList, TaskList } from '../shared/models/shared.models';
+import { Observable, Subject, filter, map, switchMap, takeUntil } from 'rxjs';
 
 @Component({
   selector: 'app-todo-list',
   templateUrl: './todo-list.component.html',
   styleUrl: './todo-list.component.scss',
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class TodoListComponent implements OnInit, OnDestroy { 
-  allTaskLists: TaskGroups | undefined;
+export class TodoListComponent implements OnInit, OnDestroy {
+  private destroy$ = new Subject<void>();
+
+  allTaskLists: TaskGroupsList | undefined;
 
   radioBtnGroup: FormGroup = new FormGroup ({
     activeTaskList: new FormControl()
   })
 
-  subscriptionToTaskList: Subscription | undefined;
+  taskList: TaskList | undefined;
 
   constructor(private apiService: ApiService, private changeDetectorRef: ChangeDetectorRef, private route: ActivatedRoute) {}

```

Comment:
`private` fields (destroy$) should be places below public ones (same for class methods)